encodings/fastlanes/src/bitpacking/compute/take.rs

```diff
@@ -35,111 +35,177 @@ impl TakeFn for BitPackedArray {
 
         let indices = indices.clone().into_primitive()?;
         let taken = match_each_unsigned_integer_ptype!(ptype, |$T| {
-            PrimitiveArray::from_vec(take_primitive::<$T>(self, &indices)?, taken_validity)
+            match_each_integer_ptype!(indices.ptype(), |$I| {
+                PrimitiveArray::from_vec(take_primitive::<$T, $I>(self, &indices)?, taken_validity)
+            })
         });
         Ok(taken.reinterpret_cast(ptype).into_array())
     }
 }
 
-fn take_primitive<T: NativePType + BitPacking>(
+// array_chunks must use while let so that we can get the remainder
+#[allow(clippy::while_let_on_iterator)]
```

Comment:
I think this is correct, you can still use the for loop and then access iterator afterwards